pkg/tracking/store/sql/experiments_search_builder.go

```diff
@@ -0,0 +1,278 @@
+							)
+						}
+
+						value = intValue
+					default:
+						return nil, contract.NewError(
+							protos.ErrorCode_INVALID_PARAMETER_VALUE,
+							fmt.Sprintf(
+								"invalid numeric attribute comparison operator '%s'", comparison,
+							),
+						)
+					}
+				case "name":
+					switch strings.ToUpper(comparison) {
+					case NotEqualExpression, EqualExpression, LikeExpression, ILikeExpression:
+						if strings.HasPrefix(value.(string), "(") {
+							return nil, contract.NewError(
+								protos.ErrorCode_INVALID_PARAMETER_VALUE,
+								fmt.Sprintf("invalid string value '%s'", value),
+							)
+						}
+
+						value = strings.Trim(value.(string), `"'`)
+
+						if database.Dialector.Name() == "sqlite" && strings.ToUpper(comparison) == ILikeExpression {
+							key = fmt.Sprintf("LOWER(%s)", key)
+							comparison = LikeExpression
+							value = strings.ToLower(value.(string))
+						}
+					default:
+						return nil, contract.NewError(
+							protos.ErrorCode_INVALID_PARAMETER_VALUE,
+							fmt.Sprintf(
+								"invalid string attribute comparison operator '%s'", comparison,
+							),
+						)
+					}
+				default:
+					return nil, contract.NewError(
+						protos.ErrorCode_INVALID_PARAMETER_VALUE,
+						fmt.Sprintf(
+							"invalid attribute '%s'. Valid values are ['name', 'creation_time', 'last_update_time']", key,
+						),
+					)
+				}
+
+				query.Where(fmt.Sprintf("%s %s ?", key, comparison), value)
+			case "tag", "tags":
+				switch strings.ToUpper(comparison) {
+				case NotEqualExpression, EqualExpression, LikeExpression, ILikeExpression:
+					if strings.HasPrefix(value.(string), "(") {
+						return nil, contract.NewError(
+							protos.ErrorCode_INVALID_PARAMETER_VALUE,
+							fmt.Sprintf("invalid string value '%s'", value),
+						)
+					}
+
+					value = strings.Trim(value.(string), `"'`)
+				default:
+					return nil, contract.NewError(
+						protos.ErrorCode_INVALID_PARAMETER_VALUE,
+						fmt.Sprintf("invalid tag comparison operator '%s'", comparison),
+					)
+				}
+
+				table := fmt.Sprintf("filter_%d", index)
+				where := fmt.Sprintf("value %s ?", comparison)
+
+				if database.Dialector.Name() == "sqlite" && strings.ToUpper(comparison) == ILikeExpression {
+					where = "LOWER(value) LIKE ?"
+					value = strings.ToLower(value.(string))
+				}
+
+				query.Joins(
+					fmt.Sprintf("JOIN (?) AS %s ON experiments.experiment_id = %s.experiment_id", table, table),
+					database.Select(
+						"experiment_id", "value",
+					).Where("key = ?", key).Where(where, value).Model(&models.ExperimentTag{}),
+				)
+			default:
+				return nil, contract.NewError(
+					protos.ErrorCode_INVALID_PARAMETER_VALUE,
+					fmt.Sprintf("invalid entity type '%s'. Valid values are ['tag', 'tags', 'attribute']", entity),
+				)
+			}
+		}
+	}
+
+	return query, nil
+}
+
+func createExperimentsNextPageToken(experiments []models.Experiment, limit, offset int) (string, *contract.Error) {
```

Comment:
I'm not sure and I'm just asking, but is this the same next token thing as in `search_runs`? Can we reuse or did it not seem worth it?